event/consumer.go

```diff
@@ -39,7 +38,6 @@ type Handler interface {
 
 // Consume converts messages to event instances, and pass the event to the provided handler.
 func (consumer *Consumer) Consume(
-	ctx context.Context,
```

Comment:
You still want to be using the context created from the service for logging purposes in `line 78` rather than creating a new context here

Reply:
yes when I though about it I think the kafka context is only needed when you add a message to batch. I got confused myself between taceid and batchid (which should be there for the entire batch). I have changed it now  so that the kafkacontext is only specific to the message consumed from kafka.